File: equipment_management/equipment_management/report/list_of_problem_reports/list_of_problem_reports.py
# Copyright (c) 2022, Deepak Kumar and contributors
# For license information, please see license.txt

import frappe
from frappe import _

def execute(filters=None):
	filters = frappe._dict(filters or {})
	columns = get_columns(filters)
	data = get_data(filters)
	return columns, data

def get_columns(filters):
	columns = [
		{
			"label": _("Equipment Inventory Number"),
			"fieldtype": "Link",
			"fieldname": "equipment",
			"options": "Equipment",
			"width": 150
		},
		{
			"label": _("Equipment Name"),
			"fieldtype": "Link",
			"fieldname": "name",
			"options": "Problem Report",
			"width": 150
		},
		{
			"label": _("Status"),
			"fieldtype": "Data",
			"fieldname": "status",
			"width": "150"
		},
		{
			"label": _("Last Location"),
			"fieldtype": "Link",
			"fieldname": "last_location",
			"options": "Location",
			"width": "150"
		},
		{
			"label": _("Location Status"),
			"fieldtype": "Data",
			"fieldname": "location_status",
			"width": "150"
		},
		{
			"label": _("Problem Title"),
			"fieldtype": "Data",
			"fieldname": "problem_title",
			"width": "150"
		},
		{
			"label": _("Posting Date"),
			"fieldtype": "Date",
			"fieldname": "posting_date",
			"width": "150"
		},
		{
			"label": _("Created By"),
			"fieldtype": "Link",
			"fieldname": "owner",
			"options": "User",
			"width": "150"
		}
	]
	return columns

def get_conditions(filters):
	conditions = ''

	if filters.name:
		conditions += f'and pr.name="{filters.name}"'
		return conditions

	if filters.status:
		conditions += f'and pr.status="{filters.status}"'

	return conditions

def get_data(filters):
	
	conditions = get_conditions(filters)
	query = f'select pr.equipment,pr.name,pr.status,pr.last_location, eq.location_status, pr.problem_title, pr.posting_date, pr.owner from `tabProblem Report` pr inner join `tabEquipment` eq on pr.equipment=eq.name where pr.equipment=pr.equipment {conditions}'
	# data = frappe.db.get_all("Problem Report", fields=['equipment','name','status','last_location','problem_title','posting_date','owner'], filters=conditions, order_by='posting_date desc')
	data = frappe.db.sql(query)

	return data

File: equipment_management/equipment_management/report/list_of_problem_reports/list_of_problem_reports.js
// Copyright (c) 2022, Deepak Kumar and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["List of Problem Reports"] = {
	"filters": [
		{
			label: "Name",
			fieldname: "name",
			fieldtype: "Link",
			options: "Problem Report",
		},
		{
			label: "Status",
			fieldname: "status",
			fieldtype: "Select",
			options: "\nOpen\nResolved",
		}
	],
	onload: function(report) {
		frappe.views.QueryReport.prototype.add_card_button_to_toolbar = () => {}
		frappe.views.QueryReport.prototype.add_chart_buttons_to_toolbar = () => {}
	},
	formatter: function (value, cell, columnDef, row) {
		
		// if (columnDef.fieldname === "name")
		// 	value  = `<a href='/app/equipment-maintenance/${value}' target=_blank> ${value} </a>`;

		if (columnDef.fieldname === "status") {
			if (value==="Open") {
                return `<span class="ellipsis" title="Status: ${value}">
                <span class="filterable indicator-pill red ellipsis" data-filter="status,=,${value}">
                    <span class="ellipsis"> ${value} </span>
                </span>
            </span>`
            }
            else if (value==="Resolved") {
                return `<span class="ellipsis" title="Status: ${value}">
                <span class="filterable indicator-pill gray ellipsis" data-filter="status,=,${value}">
                    <span class="ellipsis"> ${value} </span>
                </span>
            </span>`
            }
		}
		return value
	}
};
